libglusterfs/src/common-utils.c

```diff
@@ -3686,18 +3804,26 @@ gf_is_same_address(char *name1, char *name2)
     memset(&hints, 0, sizeof(hints));
     hints.ai_family = AF_UNSPEC;
 
-    gai_err = getaddrinfo(name1, NULL, &hints, &addr1);
-    if (gai_err != 0) {
-        gf_smsg(name1, GF_LOG_WARNING, 0, LG_MSG_GETADDRINFO_FAILED, "error=%s",
-                gai_strerror(gai_err), NULL);
-        goto out;
+    if (!p1) {
+        gai_err = getaddrinfo(name1, NULL, &hints, &addr1);
```

Comment:
Why not using `gf_dns_lookup_address_cached()` to get it from cache (or add it if not already present) ?

Reply:
Because gf_is_same_address can be call by non brick process also currently dnscache has been populated by some xlator either server or iostat.

Reply:
The problem is currently dnscache is declared only for server/io-stat xlator, in case of glusterd it won't be able to access the dnscache.We can access dnscache in this function only if it has declared on glusterfs_ctx_t.If i do declare dnscache on glusterfs_ctx_t the other issue is dnscache_ttl_sec, currently the option is configurable for server and io-stat xlator only.

Reply:
oh i missed it, you wanted to check if dnscache is valid on xlator ptr then we can use cache otherwise need to call dns call.
I shall update the same in next patch.

Reply:
> The problem is currently dnscache is declared only for server/io-stat xlator, in case of glusterd it won't be able to access the dnscache.We can access dnscache in this function only if it has declared on glusterfs_ctx_t.

Yes, that's what I think we should do. The DNS cache is a global thing, not per-xlator.

> If i do declare dnscache on glusterfs_ctx_t the other issue is dnscache_ttl_sec, currently the option is configurable for server and io-stat xlator only.

The option should be global. We already have some global options, but I'm not sure if they are shared with glusterd. If not, we could read the option from glusterd.vol for glusterd.

> oh i missed it, you wanted to check if dnscache is valid on xlator ptr then we can use cache otherwise need to call dns call.

No, I don't think this is useful. The cache should be global. We don't want to resolve the same name once for each xlator that requests it.

Reply:
I have declared dnscache as a part of glusterfs_ctx, for the time being i am using only for brick process.
Later we can use same cache for rpc_trans and glusterd also.